test/tests/functional/pbs_check_calendaring.py

```diff
@@ -0,0 +1,100 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2016 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+from tests.functional import *
+
+
+class TestCalendaring(TestFunctional):
+
+    """
+    This test suite tests if PBS scheduler calendars events correctly
+    """
+    def test_topjob_start_time(self):
+        """
+        In this test we test that the top job which gets added to the
+        calendar has estimated start time correctly set for future when
+        job history is enabled and opt_backfill_fuzzy is turned off.
```

Comment:
Is turning off opt_backfill_fuzzy necessary for doing any kind of calendaring testing? If yes, then please add a custom setUp to this testsuite and turn it off there. If not, then can you please explain why this test needs it to be turned off?

Reply:
it might also be useful to set strict_ordering in setUp.

Reply:
Well, Since there is only one test in the suite, I have all the config settings in the test case itself. Also, since the file is pretty generic, I suppose moving it to a setup may not really be helpful for tests someone might add in future.
The reason I'm turning backfill_fuzzy off is because the subjobs that are in expired state have only 10 seconds of walltime and opt_backfill_fuzzy by default (when unset) tries to simulate events at 60 seconds boundary, which means the expired subjob's end event will be  performed and running subjob's end event will also be performed (since both of them lie in this 60 second boundary), after performing both the events the third subjob's run event gets placed at the right time (which is after the running subjob's end event).
If I turn off backfill-fuzzy then we will check if the job can run or not after performing each event which will mean now third subjob's run event will be placed right after the expired subjob (which will be in the past).

Reply:
Thanks for clarifying Arun. Could you please add a comment either inline in the test, or in the header, explaining why opt_backfill_fuzzy should be turned off? I bet I won't remember this the next time I look at this test :)

I'm okay with no custom setUp()

Reply:
I'll add a comment